scenarios/pfcp-buffering.xml

```diff
@@ -6,23 +6,58 @@
         <step name="Push-Netcfg-Dbuf"
               exec="onos-netcfg ${OCI} ${UP4_ROOT}/scenarios/topo/netcfg-dbuf.json"/>
-        <group name="Pfcp-Set-Buff" requires="Push-Netcfg-Dbuf">
-            <group name="Pfcp-Push-Buff">
+        <group name="Pfcp-Set-Buff-Create" requires="Push-Netcfg-Dbuf">
+            <group name="Pfcp-Push-Buff-Create">
                 <sequential var="${ENODEB#}"
                             starts="Pfcp-Buffer-Session-Create-${#}"
-                            ends="Pfcp-Buffer-Session-Modify-${#-1}">
+                            ends="Pfcp-Buffer-Session-Create-${#-1}">
                     <step name="Pfcp-Buffer-Session-Create-${#}" requires="Push-Netcfg-Dbuf" delay="5"
                           exec="mock-smf-cmd create --session-count 5 --base ${#}0 --ue-pool 17.0.${#-1}.0/24 --enb-addr 140.0.10${#-1}.1"/>
-                    <step name="Pfcp-Buffer-Session-Modify-${#}" requires="^" delay="5"
+                </sequential>
+            </group>
+            <step name="Check-Up4-Flows-Buff-Create" requires="Pfcp-Push-Buff-Create" delay="5"
+                  exec="onos-cli-grep ${OCI} up4:read-flows 'UL flows=${PFCP_UP4_FLOWS}, DL flows=${PFCP_UP4_FLOWS}'"/>
+            <step name="Check-Flow-Rules-Buff-Create" requires="Pfcp-Push-Buff-Create" delay="5"
+                  exec="onos-check-flows ${OCI}"/>
+        </group>
+        <group name="Check-Traffic-Buff-Drop-DL" requires="Pfcp-Set-Buff-Create" delay="5">
+            <parallel var="${ENODEB#}">
+                <!-- Downlink -->
+                <step name="Downlink-Enb-Recv-Gtp-Buff-Drop-DL-${#}" requires="Pfcp-Set-Buff-Create"
+                      exec="mn-cmd ${ENODEB#} traffic.py recv-none -t 10"/>
+                <step name="Downlink-Pdn-Send-Udp-Buff-Drop-DL-${#}" requires="Pfcp-Set-Buff-Create" delay="5"
+                      exec="mn-cmd pdn traffic.py send-udp -c 10 --flow-count 5 --ue-pool 17.0.${#-1}.0/24 --enb-addr 140.0.10${#-1}.1"/>
+            </parallel>
+        </group>
+        <group name="Pfcp-Set-Buff-Modify" requires="Check-Traffic-Buff-Drop-DL">
+            <group name="Pfcp-Push-Buff-Modify">
+                <sequential var="${ENODEB#}"
+                            starts="Pfcp-Buffer-Session-Modify-${#}"
+                            ends="Pfcp-Buffer-Session-Modify-${#-1}">
+                    <step name="Pfcp-Buffer-Session-Modify-${#}" requires="Check-Traffic-Buff-Drop-DL" delay="5"
+                          exec="mock-smf-cmd modify --session-count 5 --base ${#}0 --ue-pool 17.0.${#-1}.0/24 --enb-addr 140.0.10${#-1}.1"/>
+                </sequential>
+            </group>
+            <step name="Check-Up4-Flows-Buff-Modify" requires="Pfcp-Push-Buff-Modify" delay="5"
+                  exec="onos-cli-grep ${OCI} up4:read-flows 'UL flows=${PFCP_UP4_FLOWS}, DL flows=${PFCP_UP4_FLOWS}'"/>
+            <step name="Check-Flow-Rules-Buff-Modify" requires="Pfcp-Push-Buff-Modify" delay="5"
```

Comment:
shall we verify the fwd here ?

Reply:
No, I believe there is no need to verify fwd here, we already do that in the `pfcp-forwarding` scenario.